app/common.py

```diff
@@ -3,6 +3,9 @@
 ###
 
 from abc import ABC, abstractmethod, abstractclassmethod
+from firebase_admin import auth
+from .user_api.errors import UserAuthenticationError
+from werkzeug.exceptions import Unauthorized
```

Comment:
What is `werkzeug.exceptions` we don't have something from Firebase we can use, or create something ourselves?

Reply:
To my knowledge, this Unauthorized exception seems to be only thing that allows us to handle errors in the decode_token function. If any other exception type is raised in the middleware function, it results in a 500 server error. This way it is a 401 error

Reply:
> How can I test this endpoint from the frontend?

You'll need to request an ID token from Firebase, encode it as a JWT, and make a GET request to the endpoint with the JWT in the request header. If successful, it should return plain text with the user's email